### PR TITLE
Indexer: check prev_block_hash and last block height

### PR DESCRIPTION
**Pull request by olga24912**
* sleep if we reach the last block on Near
* fix: last_block value if block not exists
* validate the prev_block hash
* remove proofs indexing
* remove unused args from check-migration

### CODE REVIEW

File: src/indexer.rs [hunk 1]
```diff
@@ -221,43 +239,59 @@ impl Indexer {
             (first_block - 1, last_block, first_block - 1)
         };
 
+        // Get `current_height`
+        let current_height = if current_height == 0 {
+            if let Some(height) = self.forward_block {
+                height
+            } else {
+                0
+            }
```
Comment (karim-en) [inline src/indexer.rs:248]: Please replace it with `self.forward_block.unwrap_or_default()`

File: src/indexer.rs [hunk 2]
```diff
@@ -221,43 +239,59 @@ impl Indexer {
...
+        } else {
+            current_height
+        };
+
         // Check, do we need fetch history data or force check from some block height
         let block = if self.force_index_from_block.is_some() || self.fetch_history {
-            if let Ok(block) = client.get_block(BlockKind::Height(num_height)).await {
+            if num_height > current_height {
+                println!(
+                    "Try to fetch block with height bigger than latest block. \
```
Comment (karim-en) [inline src/indexer.rs:257]: It sounds like a call to action, and it is not clear how to fetch the block with a bigger height.

File: src/indexer.rs [hunk 3]
```diff
@@ -221,43 +239,59 @@ impl Indexer {
...
         } else {
             current_block
         };
-        let (height, chunks) = block?;
+        let (height, chunks, block_hash, prev_block_hash) = block?;
+
+        let last_block_hash = self.data.lock().unwrap().last_block_hash;
+        if let Some(block_hash) = last_block_hash {
+            if block_hash != prev_block_hash {
+                let mut data = self.data.lock().unwrap();
```
Comment (karim-en) [inline src/indexer.rs:278]: Why do we need to update the last_block if it doesn't much? And what is the entire flow of this scenario?

**Review by karim-en — CHANGES_REQUESTED** [review]

File: src/indexer.rs (hunk 2 above)
Comment (olga24912) [reply src/indexer.rs]: fixed: https://github.com/aurora-is-near/aurora-engine-migration-tool/pull/26/commits/5de7f78f370cbf6e2a51714db19f5b80b7b1fb03

File: src/indexer.rs (hunk 3 above)
Comment (olga24912) [reply src/indexer.rs:215]: last_block is the index of the block we want to fetch now.
last_handled_block is the index of the block we have successfully fetched and processed.

We have now fetched the block with the index last_block, and it turned out that its predecessor is not last_handled_block. This means that we accidentally missed some block due to network errors. Therefore, we return the pointer to last_handled_block and try to fetch the blocks again.

**Review by karim-en — APPROVED** [review]

**Review by mrLSD — APPROVED** [review]

**Pull request merged**